processing/src/main/java/org/apache/druid/frame/channel/ComposingWritableFrameChannel.java

```diff
@@ -118,14 +123,17 @@ public void close() throws IOException
   {
     if (currentIndex < writableChannelSuppliers.size()) {
       writableChannelSuppliers.get(currentIndex).get().close();
+      convertChannelSuppliersToReadOnly(currentIndex);
       currentIndex = writableChannelSuppliers.size();
     }
   }
 
   @Override
   public boolean isClosed()
   {
-    return currentIndex == writableChannelSuppliers.size();
+    return currentIndex == writableChannelSuppliers.size() || writableChannelSuppliers.get(currentIndex)
```

Comment:
Since this class does not own the `WritableFrameChannel` which are passed by the suppliers, the passed channel might be closed, externally and hence this check .

Reply:
While discussing with @cryptoe, we had the same assumption, however on revisiting the implementation of close(), we are manually setting the index to `writableChannelSuppliers.size()` on calling it, so this extra check isn't reached / needed.

Reply:
I think if we want to check for passed channel being closed, we could check that before initialising them for writes

Reply:
I have removed the additional check for now